src/Data/EuclideanRing.purs

```diff
@@ -42,7 +42,7 @@ import Data.Semiring (class Semiring, add, mul, one, zero, (*), (+))
 -- | reason not to, `Field` types should normally use this definition of
 -- | `degree`.
 class CommutativeRing a <= EuclideanRing a where
-  degree :: a -> Int
+  degree :: a -> UInt
```

Comment:
Not sure if this is exactly necessary but it does capture the "no negative stuff" rule. An unbounded positive int type may be better.

Reply:
Yeah; I'd like to mull this over a bit. Relatedly, if we merge this, are we going to want to change other parts of core libraries to use `UInt` where appropriate, like `Data.Array.length`?

Reply:
I'm not 100% sold on it. I only started doing it because I was trying to write a `EuclidianRing` instance for `UInt` and `degree` was out of range for `Int` :sweat_smile: